File: Nikhil/python/react_agent_test.py
# react_agent_test.py
from __future__ import annotations

from react_agent import ReactAgent
from person_locator_agent import PersonLocatorAgent
from weather_agent import WeatherAgent

if __name__ == "__main__":
    agent = ReactAgent(
        id_="Orchestrator",
        description="Test ReAct Agent",
        tools=[PersonLocatorAgent(), WeatherAgent()],
        check_last_step=False,
    )

    final_step = agent.execute(
        "Determine whether the temperature in the town where Maxi is located is the same as in Copenhagen."
    )

    # Print final observation to console
    print(final_step.observation)

File: Nikhil/java/src/main/java/com/infosys/small/examples/ReactAgentTest.java
package com.infosys.small.examples;

import java.util.List;

import com.infosys.small.react.ReactAgent;

public class ReactAgentTest {

	public static void main(String[] args) throws Exception {

		ReactAgent agent = new ReactAgent("Orchestrator", "Test ReAct Agent",
				List.of(new PersonLocatorAgent(), new WeatherAgent()), false);

		agent.execute("Determine whether the temperature in the town where Maxi is located the same as in Copenhagen.");

		System.out.println("//////////////////////////////////////////////////////////////////////");

//		System.out.println(agent.getPersonality());
//		System.out.println();
//		System.out
//				.println(JsonSchema.JSON_MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(agent.getSteps()));
	} 
}
